cmake/OpenCVUtils.cmake

```diff
@@ -417,38 +417,103 @@ set(OCV_COMPILER_FAIL_REGEX
     "WARNING: unknown flag:"                    # Open64
   )
 
+# optional arguments:
+# - 'UPDATED_COMPILER_FLAGS' <a>       - variable name
+# - 'UPDATED_LINKER_FLAGS' <b>         - variable name
+# - 'TEST_FILE <c1> <c2>'              - test file
+# - 'TURN_OFF_COMPILER_FLAG <d1> <d2>' - incompatible options that shall be turned off
+# - 'TURN_OFF_LINKER_FLAG <e1> <e2>'   - incompatible options that shall be turned off
+# - 'LINKER_FLAG <f1> <f2>'            - linker flag that shall be used with the compiler flag
```

Comment:
I'd suggest rewriting help string in more traditional cmake format:
```
ocv_check_compiler_flag(<LANG> <FLAG> <RESULT> [UPDATED_COMPILER_FLAGS <var>] [UPDATED_LINKER_FLAGS <var>] [TEST_FILE <file>] [TURN_OFF_COMPILER_FLAG <flag> [<flag> ...]] [TURN_OFF_LINKER_FLAG <flag> [<flag> ...]] [LINKER_FLAG <flag> [<flag> ...]])
```

Reply:
I don't think it is a good idea to mix variable update functionality to this macro. It adds unnecessary complexity and does not correlate with macro name: `ocv_check_compiler_flag`. "_check_" usually means that nothing will be changed and macro will just return some value.

Reply:
I agree but i had to deal with its usage: this macro is not always used the same variable containing the compiler/linker flags. Would you find better to pass these variables in read only and modifying them, if necessary, outside of the macro?

Reply:
Why not add these options to `add_extra_compiler_option` macro? Or a new higher level macro/function which will be used in one or two places?

Reply:
Why haven't you make a patch that you proposed in the issue? It was much simpler.